### PR TITLE
make registrating the same level as configuring

### PR DESCRIPTION
**Pull request by franzholz**

### CODE REVIEW

File: Documentation/4-FirstExtension/7-configuring-the-plugin.rst [hunk 1]
```diff
@@ -84,7 +89,7 @@ This concludes the configuration of the plugin.
 Plugin registration: :php:`ExtensionUtility::registerPlugin()`
 ==============================================================
 
-The plugin must be registered to have it actually appear as a selectable element in the backend plugin list.
+The plugin must be registered to have it appear as a selectable element in the backend. And the frontend must get the information which controller class to call for this plugin.
```
Comment (linawolf) [inline Documentation/4-FirstExtension/7-configuring-the-plugin.rst:92]: I don't get this sentence. The frontend must get the information how?

**Review by linawolf — CHANGES_REQUESTED** [review]

File: Documentation/4-FirstExtension/7-configuring-the-plugin.rst [hunk 2]
```diff
@@ -32,7 +32,7 @@ the extension directory.
    <?php
    // Prevent script from being called directly
    
-   defined('TYPO3') or die();
+   defined('TYPO3_MODE') or die();
```
Comment (linawolf) [inline Documentation/4-FirstExtension/7-configuring-the-plugin.rst:35]: TYPO3_MODE got removed in LTS 11, defined('TYPO3') or die(); is correct

File: Documentation/4-FirstExtension/7-configuring-the-plugin.rst [hunk 3]
```diff
@@ -58,23 +58,30 @@ The php closure and anonymous function is used to encapsulate the code and its v
...
-registers the method :php:`listAction()`
-in the :php:`\T3docs\StoreInventory\Controller\StoreInventoryController`.
-All actions are cached by default. If an uncached action is required,
-an additional controller/action combination must be added as the fourth parameter.
-This array has the same format as the third parameter.
-It must list all actions whose results should not be stored in the cache.
-
-This concludes the configuration of the plugin.
 
+*  The first parameter denotes the extension name in UpperCamelCase (StoreInventory). Instead of this also the extension key in lower case with underscores can be given (store_inventory).
```
Comment (linawolf) [inline Documentation/4-FirstExtension/7-configuring-the-plugin.rst:62]: Lines should have a maximum length of 80 to fascílitate handling on Github etc. Please use linebreaks here

File: Documentation/4-FirstExtension/7-configuring-the-plugin.rst [hunk 4]
```diff
@@ -84,7 +91,7 @@ This concludes the configuration of the plugin.
 Plugin registration: :php:`ExtensionUtility::registerPlugin()`
 ==============================================================
 
-The plugin must be registered to have it actually appear as a selectable element in the backend plugin list.
+The plugin must be registered to have it appear as a selectable element in the backend. The **plugin name** must be the same as it has been used in the plugin configuration.
```
Comment (linawolf) [inline Documentation/4-FirstExtension/7-configuring-the-plugin.rst:94]: line to long

File: Documentation/4-FirstExtension/7-configuring-the-plugin.rst [hunk 5]
```diff
@@ -128,5 +139,5 @@ The next call of the page with the plugin shows the inventory as a table (figure
    Figure 4-5: The output of the inventory in the frontend
 
 This marks the end of this little Extbase extension.
-This example was intentionally simple to get you started.
+This example was intentionally simple.
```
Comment (linawolf) [inline Documentation/4-FirstExtension/7-configuring-the-plugin.rst:142]: let's not talk about "simple", this causes frustration. How about "basic"?

**Review by linawolf — CHANGES_REQUESTED** [review]

File: Documentation/4-FirstExtension/7-configuring-the-plugin.rst (hunk 2 above)
Comment (franzholz) [reply Documentation/4-FirstExtension/7-configuring-the-plugin.rst]: TYPO3_MODE is deprecated in TYPO3 11 and will be removed in TYPO3 12.
``

File: Documentation/4-FirstExtension/7-configuring-the-plugin.rst (hunk 5 above)
Comment (franzholz) [reply Documentation/4-FirstExtension/7-configuring-the-plugin.rst]: small glimpse

**Review by linawolf — APPROVED** [review]

**Pull request merged**